### PR TITLE
#5254 Fix: Different sender name issue

### LINKED ISSUE #5254: Different sender name is displayed for encrypted emails sent by the extension
Description: We received a bug report in which the receiver of an encrypted message sent using FlowCrypt sees a sender name that is different from what it should be.

For example, I send an encrypted email from my test account to my corporate account. The profile name of the test account is "Test Account", which was formerly known as "Mart Robles".

When the receiver opens the email, they see that the sender name is set to the former name instead of the current one.

Screenshot 2023-06-26 at 1 34 39 PM

Screenshot 2023-06-26 at 1 35 07 PM

The reporting user also shares a test email which confirms the following issue: https://mail.google.com/mail/u/flowcrypt.compatibility@gmail.com/#inbox/FMfcgzGsnLGRgMHPqpKsCBjgJGTDMqSB

They also added that the problem does not occur when sending an original email sent by Gmail, which I have also confirmed.

Reference:
https://mail.google.com/mail/u/human@flowcrypt.com/#inbox/FMfcgzGsnLCzRmkRSSxcGFVqLrKhMSGd

### PR DESCRIPTION
**Pull request by ioanmo226**
This PR fixed different sender name issue

close #5254 // if this PR closes an issue

----------------------------------

**Tests** _(delete all except exactly one)_:
- Difficult to test (Hard to reproduce in test environment)

--------------------------------

### To be filled by reviewers

I have reviewed that this PR... _(tick whichever items you personally focused on during this review)_:
- [x] addresses the issue it closes (if any)
- [x] code is readable and understandable
- [x] is accompanied with tests, or tests are not needed
- [x] is free of vulnerabilities
- [x] is documented clearly and usefully, or doesn't need documentation

### CODE REVIEW

**Comment by ioanmo226** [comment]
@sosnovsky In case you missed this

File: extension/js/common/api/email-provider/gmail/google.ts [hunk 1]
```diff
@@ -10,6 +10,32 @@ import { GmailRes } from './gmail-parser.js';
 import { GoogleAuth } from './google-auth.js';
 import { Serializable } from '../../../platform/store/abstract-store.js';
 
+type GoogleUserProfile = {
+  resourceName: string;
+  etag: string;
+  names: GoogleName[];
+};
```
Comment (sosnovsky) [inline extension/js/common/api/email-provider/gmail/google.ts:17]: We probably don't need this full object, let's use only needed fields and move it to [GmailRes](https://github.com/FlowCrypt/flowcrypt-browser/blob/master/extension/js/common/api/email-provider/gmail/gmail-parser.ts#L15):

```js
export type GoogleProfileNames = {
    names: {
      metadata: {
        primary: boolean;
      };
      displayName: string;
    };
  };
```

File: test/source/mock/google/google-endpoints.ts [hunk 2]
```diff
@@ -154,6 +154,27 @@ export const getMockGoogleEndpoints = (oauth: OauthMock, config: GoogleConfig |
...
+        resourceName: 'people/resourceId',
+        etag: '%EgUBAi43PRoEAQIFByIMSFMydTZETy9IEFE9',
+        names: [
+          {
+            metadata: {
+              primary: true,
+            },
+            displayName: 'First Last',
+            familyName: 'First',
+            givenName: 'LLast',
```
Comment (sosnovsky) [inline test/source/mock/google/google-endpoints.ts:171]: `LLast` -> `Last`

**Review by sosnovsky — COMMENTED** [review]
Works well 👍

File: extension/js/common/api/email-provider/gmail/gmail-parser.ts [hunk 3]
```diff
@@ -122,6 +122,19 @@ export namespace GmailRes {
...
+  };
+
+  type GoogleName = {
+    metadata: GoogleMetadata;
+    displayName: string;
+  };
+
+  type GoogleMetadata = {
+    primary?: boolean;
+  };
```
Comment (sosnovsky) [inline extension/js/common/api/email-provider/gmail/gmail-parser.ts:137]: let's merge them into single `type`:
```js
export type GoogleUserProfile = {
    names: {
      metadata: {
        primary?: boolean;
      };
      displayName: string;
    };
  };
```

**Review by sosnovsky — APPROVED** [review]
👍

**Pull request merged**